tests/test_various.rs

```diff
@@ -117,3 +118,58 @@ fn pytuple_pyclass_iter() {
     py_assert!(py, tup, "type(tup[0]).__name__ == type(tup[0]).__name__");
     py_assert!(py, tup, "tup[0] != tup[1]");
 }
+
+#[pyclass(dict)]
+struct PickleSupport {}
+
+#[pymethods]
+impl PickleSupport {
+    #[new]
+    fn new(obj: &PyRawObject) {
+        obj.init({ PickleSupport {} });
+    }
+
+    pub fn __reduce__(slf: PyRef<Self>) -> PyResult<(PyObject, Py<PyTuple>, PyObject)> {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let cls = slf.to_object(py).getattr(py, "__class__")?;
+        let dict = slf.to_object(py).getattr(py, "__dict__")?;
+        Ok((cls, PyTuple::empty(py), dict))
+    }
+}
+
+fn add_module(py: Python, module: &PyModule) {
+    py.import("sys")
+        .expect("failed to import python sys module")
+        .dict()
+        .get_item("modules")
+        .expect("failed to get python modules dictionary")
+        .downcast_mut::<PyDict>()
+        .expect("failed to turn sys.modules into a PyDict")
```

Comment:
Here we call `sys.modules` and ... can it fail?
I prefer to `unwrap` if the operation doesn't fail.